test/test-applications/integrations/dependency-libs/Samples.MultiDomainHost.App.NuGetJsonWithRedirects/Program.cs

```diff
@@ -20,34 +24,123 @@ static void InnerMethodToAllowProfilerInjection()
             // Add static dependency to Newtonsoft.Json
             var serializer = new JsonSerializer();
 
-            var url = "http://www.contoso.com/";
-
             // Add dependency on System.Net.WebClient which lives in the System assembly
             // System will always be loaded domain-neutral
-            Console.WriteLine($"[WebClient] sending request to {url}");
 
-            using (var webClient = new WebClient())
+            using (StartHttpListenerWithPortResilience("http://localhost:{0}/Samples.NuGetJsonWithRedirects/", out var url))
             {
-                webClient.Encoding = Encoding.UTF8;
+                Console.WriteLine($"[WebClient] sending request to {url}");
+
+                using (var webClient = new WebClient())
+                {
+                    webClient.Encoding = Encoding.UTF8;
+
+                    var responseContent = webClient.DownloadString(url);
+                }
 
-                var responseContent = webClient.DownloadString(url);
+                // Add dependency on System.Net.HttpMessageHandler which lives in the System.Net.Http assembly
+                // System.Net.Http can be loaded in the named AppDomain if there's a bindingRedirect enforced
+                Console.WriteLine($"[HttpClient] sending request to {url}");
+                try
+                {
+                    var client = new HttpClient();
+                    var response = client.GetAsync(url).GetAwaiter().GetResult();
+                    response.EnsureSuccessStatusCode();
+                }
+                catch
+                {
+                    // do nothing
+                }
             }
 
-            // Add dependency on System.Net.HttpMessageHandler which lives in the System.Net.Http assembly
-            // System.Net.Http can be loaded in the named AppDomain if there's a bindingRedirect enforced
-            Console.WriteLine($"[HttpClient] sending request to {url}");
-            try
+            Console.WriteLine("All done!");
+        }
+
+        private static HttpListener StartHttpListenerWithPortResilience(string uriTemplate, out string uri)
```

Comment:
Should we move `StartHttpListenerWithPortResilience()` into one of the shared test projects instead of copy/pasting it several times?

Reply:
That's what I would have normally done, but those tests seem to be all about what dependencies are loaded in what order, so I'm afraid having a shared project could break something (like, if somebody in the future adds to the shared project a dependency to an assembly that is not supposed to be loaded in those tests)

Reply:
Good point! Resolving this comment.